src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -130,10 +147,10 @@ public void Dispose()
                 // For now, rely on finalization to clean up _cancellationTokenSource's wait handle (if allocated).
                 _cancellationTokenSource.Cancel();
 
-                if (_receiver != null)
+                if (_receiver.IsValueCreated)
                 {
-                    _receiver.CloseAsync().Wait();
-                    _receiver = null;
+                    Receiver.CloseAsync().Wait();
+                    _receiver = new Lazy<MessageReceiver>(() => _messagingProvider.CreateMessageReceiver(_entityPath, _serviceBusAccount.ConnectionString));
```

Comment:
On the Disposal path, just set this to null, and add a check for null above in both places "if (_receiver != null && _receiver.IsValueCreated)". Once disposed, the ThrowIfDisposed check will prevent start from being called again. I agree on the stop path with your reinitializing the Lazy receiver. Recommend moving the Lazy creation into a private helper CreateMessageReceiver so we're not duplicating the code